File: src/main/java/de/iteratec/iteraOfficeMap/ReservationRepository.java
package de.iteratec.iteraOfficeMap;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

public interface ReservationRepository extends JpaRepository<Reservation, Long> {
    List<Reservation> findByStartDateLessThanEqualAndEndDateGreaterThanEqual(
            Long endDate, Long startDate);

    List<Reservation> findByStartDateLessThanEqualAndEndDateGreaterThanEqualAndWorkplace(
            Long endDate, Long startDate, Workplace workplace);

    Reservation findByStartDateEqualsAndWorkplace(Long startDate, Workplace workplace);

    List<Reservation> findByUsernameEqualsOrderByStartDateAsc(String username);


    Reservation findByWorkplaceAndStartDateEqualsAndEndDateEquals(Workplace workplace, Long startDate, Long endDate);

    @Transactional
    List<Reservation> deleteByWorkplaceAndStartDateAndEndDate(
            Workplace workplace, Long StartDate, Long EndDate);


}

File: src/main/java/de/iteratec/officemap/reservation/ReservationRepository.kt
package de.iteratec.officemap.reservation

import de.iteratec.officemap.workplace.Workplace
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate

interface ReservationRepository : JpaRepository<Reservation, Long> {

    fun findByStartDateLessThanEqualAndEndDateGreaterThanEqual(endDate: LocalDate, startDate: LocalDate): List<Reservation>

    fun findByStartDateLessThanEqualAndEndDateGreaterThanEqualAndWorkplace(
            endDate: LocalDate, startDate: LocalDate, workplace: Workplace): List<Reservation>

    fun findByUsernameEqualsOrderByStartDateAsc(username: String): List<Reservation>

}
